sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetTableSupport.scala

```diff
@@ -369,9 +371,6 @@ private[parquet] class MutableRowWriteSupport extends RowWriteSupport {
       case DateType => writer.addInteger(record.getInt(index))
       case TimestampType => writeTimestamp(record.getLong(index))
       case d: DecimalType =>
-        if (d.precisionInfo == None || d.precisionInfo.get.precision > 18) {
-          sys.error(s"Unsupported datatype $d, cannot write to consumer")
-        }
```

Comment:
Had an offline discussion with @yhuai but forgot to post a summary here: at last we decided not to convert unlimited decimals to `decimal(10, 0)` implicitly in #6617, because firstly we need to confirm all other parts works in a consistent way, which might introduce unexpected complexity in #6617, and secondly implicit conversion can often become a huge footgun. So let's still report an error in case of `d.precisionInfo == None` (but please throw an `AnalysisException` instead of using `sys.error`).